orangecontrib/argument/miner/miner.py

```diff
@@ -1,117 +1,155 @@
-"""
-Argument mining module.
+"""Argument mining module"""
+from ast import literal_eval
+from typing import Tuple, List
+from itertools import combinations
 
-This maps arguments into an attacking network based on topic modeling results.
-"""
+import pandas as pd
 
-import itertools
-import ast
+from orangecontrib.argument.miner.utilities import check_columns
 
-import pandas as pd
 
+def select_by_topic(data: pd.DataFrame, topic: int) -> pd.DataFrame:
+    """Select arguments mentioning the given topic.
 
-class ArgumentMiner:
-    """Argument Miner class.
+    Args:
+        data (pd.DataFrame): The argument dataframe that must contain the 'topic' column.
+        topic (int): The given topic to select.
 
-    This class do the follows:
+    Returns:
+        pd.DataFrame: Part of the original argument dataframe that only contains arguments mentioning the given topic.
     """
+    expected_cols = ["topics"]
+    check_columns(expected_cols=expected_cols, data=data)
+
+    # in the result dataframe, the index will be different from the argument_id
+    # so we keep that information in a separate column.
+    data["argument_id"] = data.index
+
+    def check_topic_included(topics: Tuple[int]) -> bool:
+        try:
+            return topic in topics
+        except TypeError as ex:
+            topics = literal_eval(str(topics))
+            if isinstance(topics, int):
+                return topic == topics
+            if isinstance(topics, tuple):
+                return topic in topics
+            raise ValueError(f"Topics not tuple: {topics}") from ex
 
```

Comment:
Would it make sense to move this to utilities as well?
This function also misses docstring info + tests; it is currently tested as part of the select_by_topic function, but it may make sense to do this additionally as well, to keep an explicit idea of the expected functionality.

Reply:
Good catch! `check_topic_included` should only be used inside `select_by_topic`, and it's defined like this because `pandas.Dataframe.apply` requires a callable as input. But I agree the code this way is not very clear. So I refined the code by implementing the same logic with a for-loop instead of using `apply`, and hopefully this is clearer now,